core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java

```diff
@@ -613,6 +621,22 @@ private DiscoveryResult discoverTestClasses() {
             }
         }
 
+        Set<String> quarkusComponentTestClasses = new HashSet<>();
+        for (AnnotationInstance a : index.getAnnotations(QUARKUS_COMPONENT_TEST)) {
+            DotName name = a.target().asClass().name();
+            quarkusComponentTestClasses.add(name.toString());
+            for (ClassInfo subclass : index.getAllKnownSubclasses(name)) {
+                quarkusComponentTestClasses.add(subclass.name().toString());
+            }
+        }
+        for (ClassInfo clazz : index.getKnownUsers(QUARKUS_COMPONENT_TEST_EXTENSION)) {
+            DotName name = clazz.name();
+            quarkusComponentTestClasses.add(name.toString());
+            for (ClassInfo subclass : index.getAllKnownSubclasses(name)) {
+                quarkusComponentTestClasses.add(subclass.name().toString());
+            }
```

Comment:
This pattern is repeated several times. We may be able to extract common code by passing in the thing we're iterating over and the array to add it to.